src/main/java/no/sikt/nva/pubchannels/channelregistrycache/db/service/CacheService.java

```diff
@@ -0,0 +1,87 @@
+package no.sikt.nva.pubchannels.channelregistrycache.db.service;
+
+import static nva.commons.core.attempt.Try.attempt;
+import java.util.List;
+import java.util.UUID;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import no.sikt.nva.pubchannels.channelregistry.ChannelType;
+import no.sikt.nva.pubchannels.channelregistrycache.CachedPublicationChannelNotFoundException;
+import no.sikt.nva.pubchannels.channelregistrycache.ChannelRegistryCacheEntry;
+import no.sikt.nva.pubchannels.channelregistrycache.ChannelRegistryCsvLoader;
+import no.sikt.nva.pubchannels.channelregistrycache.db.model.ChannelRegistryCacheDao;
+import no.sikt.nva.pubchannels.handler.PublicationChannelFetchClient;
+import no.sikt.nva.pubchannels.handler.ThirdPartyPublicationChannel;
+import nva.commons.core.Environment;
+import nva.commons.core.JacocoGenerated;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.enhanced.dynamodb.DynamoDbEnhancedClient;
+import software.amazon.awssdk.enhanced.dynamodb.DynamoDbTable;
+import software.amazon.awssdk.enhanced.dynamodb.TableSchema;
+import software.amazon.awssdk.enhanced.dynamodb.model.WriteBatch;
+import software.amazon.awssdk.services.s3.S3Client;
+
+public class CacheService implements PublicationChannelFetchClient {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CacheService.class);
+    private static final int BATCH_SIZE = 25;
+    private final DynamoDbTable<ChannelRegistryCacheDao> table;
+    private final DynamoDbEnhancedClient client;
+
+    public CacheService(DynamoDbEnhancedClient client) {
+        this.client = client;
+        this.table = client.table(new Environment().readEnv("TABLE_NAME"),
+                                  TableSchema.fromImmutableClass(ChannelRegistryCacheDao.class));
+    }
+
+    @JacocoGenerated
+    public static CacheService defaultInstance() {
+        return new CacheService(DynamoDbEnhancedClient.builder().build());
+    }
+
+    public void loadCache(S3Client s3Client) {
```

Comment:
loads csv entries to DynamoDB, it takes only few minutes to populate whole datasett.